08_project/willshen8/pkg/puppy/error.go

```diff
@@ -0,0 +1,13 @@
+package puppy
+
+import "fmt"
+
+type Error struct {
+	Message string
+	Code    int
```

Comment:
Sorry should have noticed this earlier, you probably want some constants or iota for various error codes in this file, e.g. snippet from another project
```go
// ErrCode defines the kind of error
type ErrCode uint8

// Error codes
const (
	// ErrInvalidInput for invalid user input
	ErrInvalidInput ErrCode = iota
	// ErrDuplicate is used when attempting to create an already existing entry
	ErrDuplicate
	// ErrNotFound is used when trying to access a non-existing entry
	ErrNotFound
)

func (e ErrCode) String() string {
	switch e {
	case ErrInvalidInput:
		return "invalid input"
	case ErrDuplicate:
		return "duplicate"
	case ErrNotFound:
		return "not found"
	}
	return "unknown"
}
```

Reply:
@juliaogris Sorry, I missed your comment about filenames being lowercase. I happy to stick to conventions and I have changed it accordingly. Regarding the error codes, I have done lab08 before I had reviews on my lab6 or lab7. So now I have changed in all labs. The test cases are lot more succinct and in lab 9.